ui/raidboss/data/06-ew/raid/p3n.ts

```diff
@@ -0,0 +1,224 @@
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66B8', source: 'Phoinix', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P3N Heat of Condemnation',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66B2', source: 'Phoinix' }),
+      suppressSeconds: 1,
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'P3N Darkened Fire Aoe',
+      type: 'HeadMarker',
+      netRegex: NetRegexes.headMarker({ id: ['010[C-F]'] }),
+      condition: Conditions.targetIsYou(),
+      alertText: (_data, _matches, output) => {
+        return output.text!();
+      },
+      outputStrings: {
+        text: {
+          en: 'Go to add',
+        },
+      },
+    },
+    {
+      id: 'P3N Right Cinderwing',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66B4', source: 'Phoinix', capture: false }),
+      response: Responses.goLeft(),
+    },
+    {
+      id: 'P3N Left Cinderwing',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66B5', source: 'Phoinix', capture: false }),
+      response: Responses.goRight(),
+    },
+    {
+      // Could check the log line's x y coordinate to determine from where to where it charges, npc at charge target casts 66AF?
+      id: 'P3N Trail of Condemnation',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66AD', source: 'Phoinix', capture: false }),
+      alertText: (_data, _matches, output) => {
+        return output.avoidCharge!();
+      },
+      outputStrings: {
+        avoidCharge: {
+          en: 'Avoid Charge',
+          de: 'Charge ausweichen',
+          fr: 'Évitez les charges',
+          ja: '突進避けて',
+          cn: '躲避冲锋',
+          ko: '돌진을 피하세요',
+        },
+      },
+    },
+    {
+      id: 'P3N Sunbird Spawn',
+      type: 'AddedCombatant',
+      netRegex: NetRegexes.addedCombatant({ name: 'Sunbird', capture: false }),
+      suppressSeconds: 1,
+      response: Responses.killAdds(),
+    },
+    {
+      id: 'P3N Fore/Rear Carve',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66A[78]', source: 'Sunbird', capture: false }),
+      suppressSeconds: 1,
+      alertText: (_data, _matches, output) => {
+        return output.text!();
+      },
+      outputStrings: {
+        text: {
+          en: 'Dodge Add Cleave',
+        },
+      },
+    },
+    {
+      id: 'P3N Dead Rebirth',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66A9', source: 'Phoinix', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P3N Ashen Eye',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '66AB', source: 'Sparkfledged' }),
+      alertText: (data, matches, output) => {
+        if (!data.ashenEyeDirections)
+          data.ashenEyeDirections = [];
+        switch (matches.heading) {
+          case '-3.14':
```

Comment:
Same comments as on P2N about not using exact floating point numbers for positions/headings and using a formula to round from radians to 0, 1, 2, 3 directions.

Reply:
Fixed in e32226eea0497fd9f12913a9d7136930d24c0a28, still amazing formula.